### DISCUSSION TITLE
setURI does not work on last version 2.0.0-rc1

### DISCUSSION THREAD

**Discussion by max-cl** [google/model-viewer#3788] [General]
setURI is not working in the latest version 2.0.0-rc1. Today, I realized that my code was broken, I could not change textures using setURI. And checking the versions (openbase) I saw that you have released a new version (2.0.0-rc1) 4 day ago. I have solved it using version 1.12.0.

**Comment by elalish** [comment]
Indeed, it's been removed in favor of `createTexture`, which is more reliable. It's part of the breaking changes for v2.0: https://github.com/google/model-viewer/pull/3710

**Reply by elalish** [reply]
Sorry, that prerelease version was not supposed to get picked up as `latest`! That's been fixed (see #3780), but pinning your version is still a good idea so you can update to the latest API on your own time when v2.0 comes out properly.

**Reply by plutorkhan** [reply]
Can you provide an example for usage of createTexture or any link for existing samples?

**Reply by elalish** [reply]
https://modelviewer.dev/examples/scenegraph/#createTexturesExample
